qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -709,59 +824,56 @@ def is_watched(self, vm):
         return vm.name in self.vm_names
 
 
-if 'XDG_RUNTIME_DIR' in os.environ:
-    pidfile_path = os.path.join(os.environ['XDG_RUNTIME_DIR'],
-                                'qvm-start-daemon.pid')
-else:
-    pidfile_path = os.path.join(os.environ.get('HOME', '/'),
-                                '.qvm-start-daemon.pid')
+def main():
+    """ Main function of qvm-start-daemon tool"""
 
-parser = qubesadmin.tools.QubesArgumentParser(
-    description='start GUI for qube(s)', vmname_nargs='*')
-parser.add_argument('--watch', action='store_true',
-                    help='Keep watching for further domain startups')
-parser.add_argument('--force-stubdomain', action='store_true',
-                    help='Start GUI to stubdomain-emulated VGA,'
-                         ' even if gui-agent is running in the VM')
-parser.add_argument('--pidfile', action='store', default=pidfile_path,
-                    help='Pidfile path to create in --watch mode')
-parser.add_argument('--notify-monitor-layout', action='store_true',
-                    help='Notify running instance in --watch mode'
-                         ' about changed monitor layout')
-parser.add_argument('--kde', action='store_true',
-                    help='Set KDE specific arguments to gui-daemon.')
-# Add it for the help only
-parser.add_argument('--force', action='store_true', default=False,
-                    help='Force running daemon without enabled services'
-                         ' \'guivm\' or \'audiovm\'')
+    only_if_service_enabled = ['guivm', 'audiovm']
+    enabled_services = [
+        service for service in only_if_service_enabled
+        if os.path.exists(f'/var/run/qubes-service/{service}')
+    ]
 
+    args = parser.parse_args()
 
-def main(args=None):
-    """ Main function of qvm-start-daemon tool"""
-    only_if_service_enabled = ['guivm', 'audiovm']
-    enabled_services = [service for service in only_if_service_enabled if
-                        os.path.exists('/var/run/qubes-service/%s' % service)]
-    if not enabled_services and '--force' not in sys.argv and \
-            not os.path.exists('/etc/qubes-release'):
-        print(parser.format_help())
+    if args.force or os.path.exists("/etc/qubes-release"):
+        enabled_services = only_if_service_enabled
+
+    if not enabled_services:
```

Comment:
It turned out checking this before `parser.parse_args()` was important - otherwise all qubes attempt to do `admin.vm.List` call now.